router/batchrouter/handle_async.go

```diff
@@ -560,17 +590,27 @@ func (brt *Handle) createFakeJob(jobID int64, parameters stdjson.RawMessage) *jo
 	}
 }
 
-func (brt *Handle) getReportMetrics(statusList []*jobsdb.JobStatusT, parametersMap map[int64]stdjson.RawMessage) []*utilTypes.PUReportedMetric {
+type GetReportMetricsParams struct {
+	StatusList    []*jobsdb.JobStatusT
+	ParametersMap map[int64]stdjson.RawMessage
+	JobsList      []*jobsdb.JobT
+}
+
+func (brt *Handle) getReportMetrics(params GetReportMetricsParams) []*utilTypes.PUReportedMetric {
 	reportMetrics := make([]*utilTypes.PUReportedMetric, 0)
 	connectionDetailsMap := make(map[string]*utilTypes.ConnectionDetails)
 	transformedAtMap := make(map[string]string)
 	statusDetailsMap := make(map[string]*utilTypes.StatusDetail)
 	routerWorkspaceJobStatusCount := make(map[string]int)
-	for _, status := range statusList {
+	jobsMap := make(map[int64]*jobsdb.JobT)
+	for _, job := range params.JobsList {
+		jobsMap[job.JobID] = job
+	}
```

Comment:
We should pre-allocate the map since we already know what it contains? Or we can just use
[lo](https://github.com/samber/lo) package for this.
```suggestion
	jobsMap := lo.SliceToMap(params.jobsList, func(item *jobsdb.JobT) (int64, *jobsdb.JobT) {
		return item.JobID, item
	})
```